### PR TITLE
Adding syntool package

### PR DESCRIPTION
**Pull request by pramodk**
Fixing mpi variant of highfive

fyi @ferdonline : this will be used for CI in simulation toolchain.

### CODE REVIEW

**Review by ferdonline — APPROVED** [review]
Thanks for adding the spack module!

**Comment by pramodk** [comment]
@ferdonline : do you use mpi based syntool or just serial access is sufficient?

**Comment by ferdonline** [comment]
Hi Pramod.
to my mind serial the version is enough, since each process will only read and dont need any collective operation.

File: var/spack/repos/builtin/packages/syntool/package.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Syntool(CMakePackage):
+    """SYN-TOOL provides a C++ and a python API to read / write neuron
+       connectivity informations. SYN-TOOL is designed to support large
+       connecitivy data with Billions of connections."""
```
Comment (tristan0x) [inline var/spack/repos/builtin/packages/syntool/package.py]: * s/connecitivy/connectivity/
* s/Billions/billions/

**Review by tristan0x — DISMISSED** [review]

**Comment by pramodk** [comment]
thanks @tristan0x ! 
@ferdonline : I disabled backend by default as most of the people will use serial api (I assume)

**Review by matz-e — APPROVED** [review]
LGTM

**Pull request merged**